hal/src/tim.rs

```diff
@@ -0,0 +1,51 @@
+//! Timers
+
+use embedded_hal::blocking::delay::DelayUs;
+
+use crate::pac;
+
+/// Timer for delays.
+#[derive(Debug)]
+pub struct Tim2 {
+    tim2: pac::TIM2,
+}
+
+impl Tim2 {
+    /// Constructs a new timer.
+    pub fn new(tim2: pac::TIM2, rcc: &mut pac::RCC) -> Self {
+        rcc.apb1enr1.modify(|_, w| w.tim2en().enabled());
+        rcc.apb1enr1.read(); // wait for clock to be ready
+        rcc.apb1rstr1.modify(|_, w| w.tim2rst().set_bit());
+        rcc.apb1rstr1.modify(|_, w| w.tim2rst().clear_bit());
+
+        // Prescaler for 1 microsecond delay
+        tim2.psc.write(|w| w.psc().bits(48 - 1));
+
+        tim2.cr1.write(|w| w.dir().up().cen().enabled());
+        tim2.egr.write(|w| w.ug().set_bit());
+
+        Tim2 { tim2 }
+    }
+}
+
+impl DelayUs<u32> for Tim2 {
+    fn delay_us(&mut self, us: u32) {
+        // Write Auto-Reload Register (ARR)
+        // Note: Make it impossible to set the ARR value to 0, since this
+        // would cause an infinite loop.
+        self.tim2.arr.write(|w| unsafe { w.bits(us.max(1)) });
```

Comment:
PR is here: https://github.com/rust-embedded/svd2rust/pull/554

Don't wait on that PR though, it will take a while for that change to propagate to a `stm32wl` PAC release if accepted.